### PR TITLE
176428976 showdate to strong params

### PR DESCRIPTION
**Pull request by abhinavDhulipala**
Rigorously tested the update and create for strong params. Worked with @JustinRWong  to develop a process for testing these conversions. This PR should serve as a template for future strong params validation. For documentation here are the steps we took:

1.  Print the params hash within the function in questions as such (I used cucumber but you can use anything). This is so that we can properly target the URL and avoid this: ```ActionController::UrlGenerationError, No route matches```
```ruby 
SomeController#action: create/update
   puts params.as_json # you can use a prettier like postman to read this better!
``` 
2. Before removing anything create a controller test that should fail this (as we have not yet moved from ```attr_accessible```) .remember to include superfluous parameters
```ruby 
describe SomeController do
    it 'checks strong params on create' do
        expect{post :create, params}.not_to raise_error ActiveModel::MassAssignmentSecurity::Error # should fail
    end
end
```
3. When you remove attr_accessible the it should fail this:
```ruby
    expect{post :create, params}.not_to raise_error ActiveModel::ForbiddenAttributesError  # should fail
```
4.  From here write tests to validate the necessary attributes and ensure that we have properly persisted the correct attribute/and not have the other attributes. Use the ```have_key and has_key?``` to validate params NOT ```key?```. This is because params is not an ordinary hash we believe. See this [doc](https://apidock.com/rails/ActionController/Parameters/permit) for reference. We did some experiments in ```irb``` and looked at docs to figure this out. please don't waste your time also
5. Now create a private method and use it to substitute ```params[:model_to_convert]``` with the newly created private method
I've done all of this in this PR so if you guys want more detail please see the rest of the PR

### CODE REVIEW

**Review by abhinavDhulipala — COMMENTED** [review]
I reviewed it, think it looks good

**Comment by kkhus5** [comment]
Great job!

File: spec/controllers/showdates_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        show_run_dates: {start: '"2010-01-01"', end: '"2010-01-03"'},
+        stream_until: {year: 2010, month: 1, day: 1, hour: 23, minute: 45},
+      }
+      # dummy's to be filtered out
+      @dummies = [:dummy_param_1, :dummy_param_2]
+    end
+
+    context 'validate strong params' do
+
+      it 'checks strong params on create' do
```
Comment (tommywei110) [inline spec/controllers/showdates_controller_spec.rb:37]: Perhaps, we should use more descriptive title. It seems like you are testing for the removal of attr_accessor, addition of strong_params, and create function. It might be good to separate them into three individual 'It'.

File: spec/controllers/showdates_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        last_showdate = Showdate.last
+        @dummies.each { |symb| expect(last_showdate.attributes).not_to have_key(symb), message}
+        message = 'ensure proper attribute is persisted'
+        expect(last_showdate.max_advance_sales).to eq(50000), message
+      end
+
+      it 'checks strong params on update' do
+        post :create, @mass_assignment_attack_params
+        last_sdate = Showdate.last
+        expect(last_sdate.max_advance_sales).to eq 50000
```
Comment (tommywei110) [inline spec/controllers/showdates_controller_spec.rb:53]: Since this test is just for testing the update function, we should create the object directly via ShowDate.create instead of using a post request which assumes dependency on other functions.

**Review by tommywei110 — COMMENTED** [review]
From @cscoder42, @tommywei110
**What is the goal of this PR?**
Migrate from attr_accessor to strong_params

**Were you able to view screenshots, story cards, and or test the app?**
No. Technically, we don't need to for this PR.

**Is the PR in a shape that makes it easy to review?**
The description details his approach, but should include reasons for changing individual files.


**What are the status of the tests?**
Test coverage is great in that the tests cover every line. But the coverage for edge cases is lacking and the structure of tests should be broken up to showcase happy/sad paths.


**What is the style of the code like?**
Great. 

**Is there anything missing from this PR that should be included?**
Description of the files changed.

**Feature Design: Are there any parts of this feature that are cause for concern?**
There might be nothing. Could this feature be abused by users? Does the code look safe in how it treats data? Are there any potential bugs that should be guarded against?
Not really.

---

### Overall Assessment

**What does this PR do well?**
Great! It establishes the structure for migrating other controllers and how to test them.

**What is something that could be improved with this PR?**
I think the plug in (hound) is a bit distracting. The comments on individual lines are breaking up the flow of the code. Perhaps we should declutter the PR to just include implementation and functionality instead of linting.
